app/src/org/odk/collect/android/views/ODKView.java

```diff
@@ -332,26 +330,17 @@ public void setFocus(Context context) {
     /**
      * Called when another activity returns information to answer this question.
      */
-    public void setBinaryData(Object answer) {
-        boolean set = false;
-        for (QuestionWidget q : widgets) {
-            if (q instanceof IBinaryWidget) {
-                if (((IBinaryWidget) q).isWaitingForBinaryData()) {
-                    ((IBinaryWidget) q).setBinaryData(answer);
-                    set = true;
-                    break;
-                }
-            }
+    public void setBinaryData(Object answer, PendingCalloutInterface pendingCalloutInterface) {
+        FormIndex questionFormIndex = pendingCalloutInterface.getPendingCalloutFormIndex();
+        if (questionFormIndex == null) {
+            return;
         }
 
-        if (!set) {
-            Log.w(TAG, "Attempting to return data to a widget or set of widgets not looking for data");
-                     
-            for (QuestionWidget q : widgets) {
-                if (q instanceof IBinaryWidget) {
-                    ((IBinaryWidget) q).setBinaryData(answer);
-                    break;
-                }
```

Comment:
Don't understand why this `!set` conditional is here. I took it out, is that wrong?